### DISCUSSION TITLE
What language is Scriptkit written in?

### DISCUSSION THREAD

**Discussion by stevebarakat** [johnlindquist/kit#443] [Q&A] [answered]
What language is Scriptkit written in?

**Comment by johnlindquist** [comment, marked as answer]
### Short answer: TypeScript

### Medium answer:

Script Kit spans 3 projects:

> Getting a dev build working is still a pretty complicated, but once I simplify it more, I'll share how people can contribute:

1. "kit": https://github.com/johnlindquist/kit/tree/alpha
"kit" (this repo) is the "core" of Script Kit. It includes all the tools that allow you to create/run/etc scripts in the terminal/etc. It also has some AppleScript wrappers for system tools. The system tools will eventually be phased into Rust, so please don't bother messing with them. There are some bash scripts as well for build/etc

2. "Kit.app": https://github.com/johnlindquist/kitapp/tree/alpha
"Kit.app" communicates with "kit" through forking node processes.

3. scriptkit.com is currently private, but it uses a custom protocol handler for installing scripts to "Kit.app"

"4". "kit" will generate bash scripts that wrap the scripts in your "kenv" (_k_ it _env_ ironment) to make them easier to run in the terminal

### Long answer

I'll eventually write extensive docs/lessons on how it works
